crates/solver/src/liquidity/order_converter.rs

```diff
@@ -65,6 +66,30 @@ impl OrderConverter {
     }
 }
 
+/// Builds a "synthetic" order with adjusted `sell_amount` and `fee_amount` in case the order
+/// is a limit order. Returns the original order for other order classes.
+fn build_synthetic_order_if_limit_order(mut order: Order) -> Result<Order> {
```

Comment:
Shouldn't these synthetic values be set in the `LimitOrder` instance and not modify the original `Order`? The reason I am suggesting this is that we need the original order data for encoding the trade for the `settle` calldata. Otherwise, we would need special `if order.metadata.class == OrderClass::Limit` when encoding trades as well.

Reply:
That's a good point. I wrote it like this to make it a small, self-contained change. But requiring another `if` in the encoding logic is indeed not nice.

Reply:
My proposal can still be self-contained right? We just need to effectively:

```rust
let (sell_amount, buy_amount) = compute_synthetic_order_amounts_if_limit_order(...);
```

Instead of pre-mutating the order. Or is the `order.data.sell_amount` used for important things in other places as well?

Reply:
I actually already started implementing it like you suggested in the mean time.
> Or is the order.data.sell_amount used for important things in other places as well?

Not sure; have to check. 👀